### PR TITLE
[move-coverage]: change visibility for uncovered locations

### PR DESCRIPTION
**Pull request by Rqnsom**
## Description
In the `move-mutation-test` tool, the `uncovered_locations` from `SourceCoverageBuilder` are going to be used to generate mutants for specific pieces of source code that hasn't been covered by the unit test coverage.

## How Has This Been Tested?
N/A

## Key Areas to Review
N/A

## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [x] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [x] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Move Compiler
- [ ] Other (specify)

## Checklist
- [x] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I identified and added all stakeholders and component owners affected by this change as reviewers
- [x] I tested both happy and unhappy path of the functionality
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: third_party/move/tools/move-coverage/src/source_coverage.rs [hunk 1]
```diff
@@ -27,15 +27,23 @@ use std::{
     str::FromStr,
 };
 
+/// Function source coverage.
```
Comment (vineethk) [inline third_party/move/tools/move-coverage/src/source_coverage.rs:30]:
```suggestion
/// Source-level coverage information for a function.
```

File: third_party/move/tools/move-coverage/src/source_coverage.rs [hunk 2]
```diff
@@ -27,15 +27,23 @@ use std::{
...
 #[derive(Clone, Debug, Serialize)]
 pub struct FunctionSourceCoverage {
+    /// Indication for native functions.
```
Comment (vineethk) [inline third_party/move/tools/move-coverage/src/source_coverage.rs:33]:
```suggestion
    /// Is this a native function?
    /// If so, then `uncovered_locations` is empty.
```

File: third_party/move/tools/move-coverage/src/source_coverage.rs [hunk 3]
```diff
@@ -27,15 +27,23 @@ use std::{
...
     pub fn_is_native: bool,
+
+    /// List of uncovered locations in the function.
```
Comment (vineethk) [inline third_party/move/tools/move-coverage/src/source_coverage.rs:36]:
```suggestion
    /// List of source locations in the function that were not covered.
```

Comment (vineethk) [reply third_party/move/tools/move-coverage/src/source_coverage.rs:36]: I haven't looked at the code deeply, I am assuming this is what it means from looking here. Please double check.

File: third_party/move/tools/move-coverage/src/source_coverage.rs [hunk 4]
```diff
@@ -27,15 +27,23 @@ use std::{
...
     pub uncovered_locations: Vec<Loc>,
 }
 
+/// Builder for the source code coverage.
 #[derive(Debug, Serialize)]
 pub struct SourceCoverageBuilder<'a> {
-    uncovered_locations: BTreeMap<Identifier, FunctionSourceCoverage>,
+    /// List of uncovered locations in the source code.
```
Comment (vineethk) [inline third_party/move/tools/move-coverage/src/source_coverage.rs:43]: Mapping from function name to the source-level uncovered locations for that function.

File: third_party/move/tools/move-coverage/src/source_coverage.rs [hunk 5]
```diff
@@ -27,15 +27,23 @@ use std::{
...
+    pub uncovered_locations: BTreeMap<Identifier, FunctionSourceCoverage>,
+
+    /// Source map.
```
Comment (vineethk) [inline third_party/move/tools/move-coverage/src/source_coverage.rs:46]: Redundant comment, remove.

File: third_party/move/tools/move-coverage/src/source_coverage.rs (hunk 1 above)
Comment (Rqnsom) [reply third_party/move/tools/move-coverage/src/source_coverage.rs]: Done

File: third_party/move/tools/move-coverage/src/source_coverage.rs (hunk 5 above)
Comment (Rqnsom) [reply third_party/move/tools/move-coverage/src/source_coverage.rs]: Done

File: third_party/move/tools/move-coverage/src/source_coverage.rs (hunk 4 above)
Comment (Rqnsom) [reply third_party/move/tools/move-coverage/src/source_coverage.rs]: Done

File: third_party/move/tools/move-coverage/src/source_coverage.rs (hunk 3 above)
Comment (Rqnsom) [reply third_party/move/tools/move-coverage/src/source_coverage.rs]: Yes, that's right! I'm using this in the `move-mutation-test` a lot these days.
Done

File: third_party/move/tools/move-coverage/src/source_coverage.rs (hunk 2 above)
Comment (Rqnsom) [reply third_party/move/tools/move-coverage/src/source_coverage.rs]: Done

**Review by vineethk — APPROVED** [review]

**Review by GotenJBZ — APPROVED** [review]

**Review by gwapIgot — APPROVED** [review]

**Pull request merged**